atlp-rwanda/e-commerce-serge-30-bn
# chore(swagger-jsdoc): This commit includes the necessary configuratio…

### What does this PR do?
This PR sets up Swagger documentation for our API using JSDoc.

### Description of Task to be completed?
The task involved setting up Swagger documentation for our API. This was achieved by installing necessary dependencies such as swagger-jsdoc and swagger-ui-express, creating a new folder called docs in src, and included swagger.ts file as the main file with jsdoc configurations, documentation.ts as file with all endpoint's annotations with openApi, and swaggerDark.css for styling the ui.

### How should this be manually tested?
To manually test this, run the application and navigate to the /docs path in your web browser. You should be able to see the interactive API documentation generated by Swagger.

### Any background context you want to provide?
The need for this setup arose from the requirement to have a standardized and interactive documentation for our API. Swagger, combined with JSDoc, provides a robust solution for this.

### What are the relevant pivotal tracker/Trello stories?
#187364839.
